### ISSUE TITLE
[Bug]: Collection load, queryNodes offline, Restart queryNodes, Collection load and get_query_segment_info not worked

### ISSUE THREAD

**Issue by siwenyang** [milvus-io/milvus#18025]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: v2.0.1
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): v2.0.1
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: deployed on aws ec2s with terraform and anisble, have 3 c5.4 xlarge as queryNodes
```

### Current Behavior

1. Already had 50 collections(all less than 24 million), Insert 24 million data into a new collection, had 3 queryNodes, each on 32GB c5.4 xlarge
2. build collection index
3. use collection.load(), not worked:

> RPC error: [wait_for_loading_collection], <BaseException: (code=1, message=collection 434293026370617345 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2022-07-02 22:02:26.657550', 'RPC error': '2022-07-02 22:10:26.138934'}>
RPC error: [load_collection], <BaseException: (code=1, message=collection 434293026370617345 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2022-07-02 18:50:49.382295', 'RPC error': '2022-07-02 22:10:26.139401'}>
Traceback (most recent call last):
  File "search_engine.py", line 476, in <module>
    collection.load()
  File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/orm/collection.py", line 457, in load
    conn.load_collection(self._name, timeout=timeout, **kwargs)
  File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/decorators.py", line 56, in handler
    raise e
  File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/decorators.py", line 41, in handler
    return func(self, *args, **kwargs)
  File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/decorators.py", line 74, in handler
    raise e
  File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py", line 669, in load_collection
    self.wait_for_loading_collection(collection_name, timeout)
  File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/decorators.py", line 56, in handler
    raise e
  File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/decorators.py", line 41, in handler
    return func(self, *args, **kwargs)
  File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/decorators.py", line 74, in handler
    raise e
  File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py", line 689, in wait_for_loading_collection 
    return self._wait_for_loading_collection_v2(collection_name, timeout)
  File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py", line 713, in _wait_for_loading_collection_v2
    raise BaseException(response.status.error_code, response.status.reason)
pymilvus.client.exceptions.BaseException: <BaseException: (code=1, message=collection 434293026370617345 has not been loaded to memory or load failed)>

5. query nodes offline
6. restart query nodes
7. try collection.load(), not worked
8.  utility.get_query_segment_info() on other collections. not worked:

> seg = utility.get_query_segment_info(TABLE_NAME)
> RPC error: [get_query_segment_info], <BaseException: (code=1, message=getSegmentInfo: queryNode 33 is offline)>, <Time:{'RPC start': '2022-07-02 22:56:50.987348', 'RPC error': '2022-07-02 22:56:51.014065'}>
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/orm/utility.py", line 574, in get_query_segment_info
>     return _get_connection(using).get_query_segment_info(collection_name, timeout=timeout)
>   File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/decorators.py", line 56, in handler
>     raise e
>   File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/decorators.py", line 41, in handler
>     return func(self, *args, **kwargs)
>   File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/decorators.py", line 74, in handler
>     raise e
>   File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/decorators.py", line 70, in handler
>     return func(*args, **kwargs)
>   File "/home/siwen/.conda/envs/aencoder/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py", line 468, in get_query_segment_info
>     raise BaseException(status.error_code, status.reason)
> pymilvus.client.exceptions.BaseException: <BaseException: (code=1, message=getSegmentInfo: queryNode 33 is offline)>

9. But actually I already restarted queryNodes:

> / # ETCDCTL_API=3 etcdctl get --prefix by-dev/meta/session 
by-dev/meta/session/datacoord
{"ServerID":23,"ServerName":"datacoord","Address":"172.31.11.39:13333","Exclusive":true,"TriggerKill":true}
by-dev/meta/session/datanode-24
{"ServerID":24,"ServerName":"datanode","Address":"172.31.55.120:21124","TriggerKill":true}
by-dev/meta/session/id
37
by-dev/meta/session/indexcoord
{"ServerID":3,"ServerName":"indexcoord","Address":"172.31.0.12:31000","Exclusive":true,"TriggerKill":true}
by-dev/meta/session/indexnode-25
{"ServerID":25,"ServerName":"indexnode","Address":"172.31.50.224:21121","TriggerKill":true}
by-dev/meta/session/proxy-10
{"ServerID":10,"ServerName":"proxy","Address":"172.31.51.79:19530","TriggerKill":true}
by-dev/meta/session/querycoord
{"ServerID":32,"ServerName":"querycoord","Address":"172.31.14.88:19531","Exclusive":true,"TriggerKill":true}
by-dev/meta/session/querynode-34
{"ServerID":34,"ServerName":"querynode","Address":"172.31.53.14:21123","TriggerKill":true}
by-dev/meta/session/querynode-35
{"ServerID":35,"ServerName":"querynode","Address":"172.31.59.50:21123","TriggerKill":true}
by-dev/meta/session/querynode-36
{"ServerID":36,"ServerName":"querynode","Address":"172.31.55.248:21123","TriggerKill":true}
by-dev/meta/session/rootcoord
{"ServerID":1,"ServerName":"rootcoord","Address":"172.31.1.175:53100","Exclusive":true,"TriggerKill":true}

### Milvus Log

[query_coord1_new.log](https://github.com/milvus-io/milvus/files/9034169/query_coord1_new.log)
[query_node1_new.log](https://github.com/milvus-io/milvus/files/9034170/query_node1_new.log)
[proxy_new.log](https://github.com/milvus-io/milvus/files/9034171/proxy_new.log)

**Comment by siwenyang** [comment]
Also I try drop collection by utility.drop_collection(), also not worked. 
It just hanging there forever no response

**Comment by yanliang567** [comment]
@siwenyang thank you for the issue, Could you please refer this [script](https://github.com/milvus-io/milvus/tree/master/deployments/export-log) to export the whole Milvus logs for investigation?

/assign @siwenyang 
/unassign

**Comment by xiaofan-luan** [comment]
pls take a look on it

**Comment by yah01** [comment]
Hey @siwenyang, are these the full logs? The QueryCoord log seems been truncated, and no problem in it, or could you tell me how you reproduce this?

**Comment by siwenyang** [comment]
> 

@yah01 QueryCoord log is truncated, some old parts are ignored.

To reproduce this:
1. collection.load(), load several collections to fill-up query node memory usage
2. without collection.release(), try load another big collection, cause all query nodes offline
3. restart all query nodes, will need some time for pymilvus collection or utility functions to see the new query nodes online

My main confusion is why load collection to full query nodes would cause all query nodes offline, and I got no warning or logs from query nodes.

Thank you

**Comment by yah01** [comment]
@siwenyang 
The QueryNode checks whether the free memory is enough to load collection, but the prediction may be not precise, that's why the case exists.
As I known, this is a known bug which will be fixed in version 2.1, more details: issue #16887 and its related PRs
